models/glif_psc.cpp

```diff
@@ -0,0 +1,644 @@
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  NEST is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "glif_psc.h"
+
+// C++ includes:
+#include <iostream>
+#include <limits>
+
+// Includes from libnestutil:
+#include "numerics.h"
+#include "propagator_stability.h"
+
+// Includes from nestkernel:
+#include "exceptions.h"
+#include "kernel_manager.h"
+#include "name.h"
+#include "universal_data_logger_impl.h"
+
+// Includes from sli:
+#include "dict.h"
+#include "dictutils.h"
+#include "doubledatum.h"
+#include "integerdatum.h"
+#include "lockptrdatum.h"
+
+using namespace nest;
+
+nest::RecordablesMap< nest::glif_psc > nest::glif_psc::recordablesMap_;
+
+namespace nest
+{
+// Override the create() method with one call to RecordablesMap::insert_()
+// for each quantity to be recorded.
+template <>
+void
+RecordablesMap< nest::glif_psc >::create()
+{
+  insert_( names::V_m, &nest::glif_psc::get_V_m_ );
+  insert_( names::ASCurrents_sum, &nest::glif_psc::get_ASCurrents_sum_ );
+  insert_( names::I, &nest::glif_psc::get_I_ );
+  insert_( names::I_syn, &nest::glif_psc::get_I_syn_ );
+  insert_( names::threshold, &nest::glif_psc::get_threshold_ );
+  insert_( names::threshold_spike, &nest::glif_psc::get_threshold_spike_ );
+  insert_( names::threshold_voltage, &nest::glif_psc::get_threshold_voltage_ );
+}
+}
+
+/* ----------------------------------------------------------------
+ * Default constructors defining default parameters and state
+ * ---------------------------------------------------------------- */
+
+nest::glif_psc::Parameters_::Parameters_()
+  : G_( 9.43 )               // in nS
+  , E_L_( -78.85 )           // in mV
+  , th_inf_( -51.68 - E_L_ ) // in mv, rel to E_L_, - 51.68 - E_L_, i.e., 27.17
+  , C_m_( 58.72 )            // in pF
+  , t_ref_( 3.75 )           // in ms
+  , V_reset_( 0.0 )          // in mV, rel to E_L_, -78.85 - E_L_
+  , th_spike_add_( 0.37 )    // in mV
+  , th_spike_decay_( 0.009 ) // in 1/ms
+  , voltage_reset_fraction_( 0.20 )
+  , voltage_reset_add_( 18.51 )                          // in mV
+  , th_voltage_index_( 0.005 )                           // in 1/ms
+  , th_voltage_decay_( 0.09 )                            // in 1/ms
+  , asc_init_( std::vector< double >( 2, 0.0 ) )         // in pA
+  , asc_decay_( std::vector< double >{ 0.003, 0.1 } )    // in 1/ms
+  , asc_amps_( std::vector< double >{ -9.18, -198.94 } ) // in pA
+  , asc_r_( std::vector< double >( 2, 1.0 ) )
+  , tau_syn_( 1, 2.0 ) // in ms
```

Comment:
I'm a little confused over why there are two synapses by default. Is this just by means of example?

Unrelated to that, I'm surprised that an std::vector can be initialised this way. Could you make it `tau_syn_( std::vector< double >(1, 2) )` for consistency?

Reply:
For glif_psc, it has one default synapse with time constant as 2.0ms. For glif_cond, it has two default synapses, one as excitation and one as inhibition. For both models, users can set as many synapses as they needed. I have changed the `tau_syn_` stating.